src/flows/components/header/index.tsx

```diff
@@ -23,12 +26,40 @@ import {serialize} from 'src/flows/context/flow.list'
 import {FeatureFlag} from 'src/shared/utils/featureFlag'
 import {updatePinnedItemByParam} from 'src/shared/contexts/pinneditems'
 import {getOrg} from 'src/organizations/selectors'
+import {getAuthorizations} from 'src/client/generatedRoutes'
+import {RemoteDataState} from 'src/types'
 
-const FULL_WIDTH = true
+interface Token {
+  token: string
+  description: string
+}
+
+interface Share {
+  id: string
+  accessID: string
+}
 
 const FlowHeader: FC = () => {
   const {flow, updateOther, id} = useContext(FlowContext)
   const {id: orgID} = useSelector(getOrg)
+  const [sharing, setSharing] = useState(false)
+  const [token, setToken] = useState<Token>()
+  const [loadingToken, setLoadingToken] = useState(RemoteDataState.NotStarted)
+  const [tokens, setTokens] = useState<Token[]>([])
+  const [share, setShare] = useState<Share>()
+  const [linkLoading, setLinkLoading] = useState(RemoteDataState.NotStarted)
+
+  useEffect(() => {
+    // TODO: once we have a generated func from swagger, use that
+    fetch(`/api/v2private/notebooks/share?notebookID=${id}`).then(res =>
```

Comment:
is this todone?

Reply:
nah I need to update open api first. Can do that shortly